src/pointer-analysis/value_set.cpp

```diff
@@ -414,7 +415,7 @@ void value_sett::get_value_set_rec(
 
     default:
       log_error("Unexpected side-effect: {}", *expr);
-      abort();
+      throw std::runtime_error("unsuported side-effect");
```

Comment:
If this exception is never caught we should continue to abort().

If it is caught and handled somewhere then an own `value_set_error` or similar exception class would make the usage clearer.

Reply:
Added a class for the exception

Reply:
Nice, thanks.